File: iosApp/iosApp/ViewController/Login/Mapper/SmartSaverTransactionPresentationToUiModelMapper.swift
//
//  SmartSaverTransactionPresentationToUiModelMapper.swift
//  iosApp
//
//  Created by Animasahun Ibrahim on 17/03/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

class SmartSaverTransactionPresentationToUiModelMapper: PresentationToUiMapper{
    typealias T = SmartSaverTransactionPresentationModel
    
    typealias U = SmartSaverTransactionUiModel
    
    func toUi(input: SmartSaverTransactionPresentationModel) -> SmartSaverTransactionUiModel {
        
        return .init(
            amount: input.amount,
            type: input.type,
            narration: input.narration,
            dateCreated: input.dateCreated
        )
    }
    
}

File: app-ui/src/main/java/com/teewhydope/app/ui/login/mapper/SmartSaverTransactionPresentationToUiModelMapper.kt
package com.teewhydope.app.ui.login.mapper

import com.teewhydope.app.common.mapper.PresentationToUiMapper
import com.teewhydope.app.presentation.login.model.SmartSaverTransactionPresentationModel
import com.teewhydope.app.ui.login.model.SmartSaverTransactionUiModel

class SmartSaverTransactionPresentationToUiModelMapper :
    PresentationToUiMapper<SmartSaverTransactionPresentationModel, SmartSaverTransactionUiModel>() {
    override fun map(input: SmartSaverTransactionPresentationModel) =
        SmartSaverTransactionUiModel(
            amount = input.amount,
            type = input.type,
            narration = input.narration,
            dateCreated = input.dateCreated,
        )
}